### PR TITLE
[test] Add rom_e2e_default_otp_bootup test

### LINKED ISSUE #16352: [test] Add a ROM e2e test with default OTP values (all zeroes) in TEST_UNLOCK with execution enabled.
This reminds me: we should probably add a simple boot test with all default OTP values in TEST_UNLOCKED.

_Originally posted by @alphan in https://github.com/lowRISC/opentitan/issues/16192#issuecomment-1315136765_

### PR DESCRIPTION
**Pull request by milesdai**
Closes #16352

### CODE REVIEW

**Review by alphan — APPROVED** [review]
Thanks @milesdai !

File: sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson [hunk 1]
```diff
@@ -18,6 +18,22 @@
...
+      desc: '''Verify that ROM can boot a ROM_EXT with default OTP values (all zeroes).
+
+            - Create an OTP with all zeros except:
+                - OWNER_SW_CFG_ROM_BOOTSTRAP_EN set to 0x739 (true)
+                - CREATOR_SW_CFG_ROM_EXEC_EN set to 0xffffffff (enabled)
+                - LC_STATE in the TEST_UNLOCKED0 state
+                - LIFE_CYCLE count value set to 1
+            - Ensure that the ROM can boot.
+            '''
+      tags: ["rom", "verilator", "dv", "fpga", "silicon"]
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:32]: was this supposed to be run in DV or Verilator? from the looks of the BUILD test target below, I would assume no? if not then this should read:
```
tags: ["rom", "no_dv", "fpga", "silicon"]
```

As is, it is showing up in the DV nightly dashboard as failing.

File: sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson [hunk 2]
```diff
@@ -18,6 +18,22 @@
...
+      stage: V2
+      tests: ["rom_e2e_default_otp_smoke"]
```
Comment (timothytrippel) [inline sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:34]: Also, this `tests` list should be empty if it the test is not in DV.

File: sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson (hunk 1 above)
Comment (milesdai) [reply sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:32]: Argh sorry about that -- I moved a bit too fast on this PR. I submitted #16704 to fix.

Comment (milesdai) [reply sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:32]: @alphan Just to confirm, does this test need to be run in DV or Verilator at the moment?

Comment (timothytrippel) [reply sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:32]: no problem, thanks for the fix!

Comment (alphan) [reply sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson:32]: @milesdai no, it doesn't need to be run in DV or verilator.

**Pull request merged**